src/Raven.Server/Documents/Indexes/Static/MismatchedReferencesHandler.cs

```diff
@@ -0,0 +1,82 @@
+using System.Collections.Generic;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Indexes.Static;
+
+public class MismatchedReferencesHandler
```

Comment:
I wonder if `MismatchedReferencesWarningHandler` wouldn't be more suitable name. It would explain that it's about warnings only since it doesn't actually handle / fix / correct the mismatched loads - it just collects them so we can generate a warning based on that.

Reply:
Another name I can think of  is `MismatchedReferencesCollector` or `MismatchedLoadDocumentCollector`

Reply:
Done